eng/scripts/Get-TypeSpec-Folders.ps1

```diff
@@ -9,8 +9,9 @@ param (
 )
 
 $tspFiles = @()
+$osSlash = [IO.Path]::DirectorySeparatorChar
 if ([string]::IsNullOrEmpty($TargetBranch) -or [string]::IsNullOrEmpty($SourceBranch)) {
-  $tspFiles = (Get-ChildItem -path ./specification tspconfig.yaml -Recurse).FullName -replace "$($pwd.Path)/"
+  $tspFiles = (Get-ChildItem -path ./specification tspconfig.yaml -Recurse).FullName -replace [Regex]::Escape("$($pwd.Path)$osSlash")
 }
 else {
   Write-Host "git -c core.quotepath=off -c i18n.logoutputencoding=utf-8 diff --name-only `"$TargetBranch...$SourceBranch`" -- | Where-Object {`$_.StartsWith('specification')}"
```

Comment:
For this case do we want to scope it to *.tsp or we checking for any files?

Reply:
Our heuristic is if **any** file is changed under folder `specification/foo`, we will validate **all** TypeSpec specs under folder `specification/foo`.

It's not sufficient to only check for changes to `*.tsp` files, since changes to `tspconfig.yaml` or AutoRest `*.json` can also cause validation to fail.

It's also not sufficient to only validate specs if a file changed under the folder containing `tspconfig.yaml`, since specs can import other files from sibling folders (and maybe even parent folders).  But we do have guidance for the repo that specs should not import files outside their `specfication/foo` folder.

This may result in validating more specs than are strictly necessary for a PR, but we'd rather err on the side of validating too much than too little.  All specs in main should be passing at all times, since if any spec is failing in main it should be fixed ASAP.